Algorithms/JavaScript/all-anagrams.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Given a single input string, write a function that produces all possible anagrams
+ * of a string and outputs them as an array. At first, don't worry about
+ * repeated strings.  What time complexity is your solution?
+ *
+ * Extra credit: Deduplicate your return array without using uniq().
+ */
+
+/**
+  * example usage:
+  * var anagrams = allAnagrams('abc');
+  * console.log(anagrams); // [ 'abc', 'acb', 'bac', 'bca', 'cab', 'cba' ]
+  */
+
+var allAnagrams = function(string) {
+
+  if (typeof(string) !== 'string') {return;}
```

Comment:
nitpick: I'd remove the curlies and (personally, but optionally) put return on the next line